Source/WebCore/platform/graphics/FontDescription.h

```diff
@@ -149,6 +176,27 @@ class FontDescription {
     void setVariantEastAsianWidth(FontVariantEastAsianWidth variant) { m_variantEastAsianWidth = enumToUnderlyingType(variant); }
     void setVariantEastAsianRuby(FontVariantEastAsianRuby variant) { m_variantEastAsianRuby = enumToUnderlyingType(variant); }
     void setVariantEmoji(FontVariantEmoji variant) { m_variantEmoji = enumToUnderlyingType(variant); }
+    void setVariantEastAsian(FontVariantEastAsianValues values)
+    {
+        setVariantEastAsianVariant(values.variant);
+        setVariantEastAsianWidth(values.width);
+        setVariantEastAsianRuby(values.ruby);
+    }
+    void setVariantNumeric(FontVariantNumericValues values)
+    {
+        setVariantNumericFigure(values.figure);
+        setVariantNumericSpacing(values.spacing);
+        setVariantNumericFraction(values.fraction);
+        setVariantNumericOrdinal(values.ordinal);
+        setVariantNumericSlashedZero(values.slashedZero);
+    }
+    void setVariantLigatures(FontVariantLigaturesValues values)
+    {
+        setVariantCommonLigatures(values.common);
+        setVariantDiscretionaryLigatures(values.discretionary);
+        setVariantHistoricalLigatures(values.historical);
+        setVariantContextualAlternates(values.contextual);
+    }
```

Comment:
I would put these vertically formatted function bodies outside the class definition.